From f128c901210e6b7ec5ff92e3b48d486991a3abdc Mon Sep 17 00:00:00 2001
From: Maruthi Srinivas Bayyavarapu <maruthi.srinivas.bayyavarapu@xilinx.com>
Date: Thu, 16 Aug 2018 16:53:20 +0530
Subject: [PATCH 0143/1852] dt-bindings: ASoC: Document audio formatter
 bindings for xlnx,audio-formatter

commit 6714745af33a11ff5bd304d59e169b1158c12681 from
https://github.com/Xilinx/linux-xlnx.git

Added documentation for audio formatter IP core DT bindings.

Updated documentation of connected source and sink nodes. These nodes
will be parsed and if successful sound card device is created. Parsed
nodes are passed as platform data for sound card driver. Those nodes
will be used to identify the DAI link to be used in sound card driver.

Added clocks. s_axi_lite_aclk will be used and enabled/disabled in
audio formatter driver. aud_mclk will be used in PL sound card driver
to dynamically change the mclk connected to various audio interfaces.
This change adds audio use case at different sampling rates.

Signed-off-by: Maruthi Srinivas Bayyavarapu <maruthi.srinivas.bayyavarapu@xilinx.com>
Reviewed-by: Hyun Kwon <hyun.kwon@xilinx.com>
Signed-off-by: Michal Simek <michal.simek@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 .../bindings/sound/xlnx,audio-formatter.txt     | 17 ++++++++++++++++-
 1 file changed, 16 insertions(+), 1 deletion(-)

diff --git a/Documentation/devicetree/bindings/sound/xlnx,audio-formatter.txt b/Documentation/devicetree/bindings/sound/xlnx,audio-formatter.txt
index cbc93c8f4963..e98ed1706e21 100644
--- a/Documentation/devicetree/bindings/sound/xlnx,audio-formatter.txt
+++ b/Documentation/devicetree/bindings/sound/xlnx,audio-formatter.txt
@@ -1,6 +1,6 @@
 Device-Tree bindings for Xilinx PL audio formatter
 
-The IP core supports DMA, data formatting(AES<->PCM conversion)
+The IP core supports DMA, data formatting(packing, conversion)
 of audio samples.
 
 Required properties:
@@ -13,6 +13,19 @@ Required properties:
  - interrupts-parent: Phandle for interrupt controller.
  - interrupts: List of Interrupt numbers.
  - reg: Base address and size of the IP core instance.
+ - xlnx,tx: connected audio sink node.
+	    Should be one of below supported nodes:
+		1. HDMI video Tx output
+		2. I2S transmitter
+		3. UHDSDI audio embed
+	    Only those nodes were supported in sound card driver
+
+ - xlnx,rx: connected audio source node.
+	    Should be one of below supported nodes:
+		1. HDMI video Rx input
+		2. I2S receiver
+		3. UHDSDI audio extract
+	    Only those nodes were supported in sound card driver
  - clock-names: List of input clocks.
    Required elements: "s_axi_lite_aclk", "aud_mclk"
  - clocks: Input clock specifier. Refer to common clock bindings.
@@ -24,6 +37,8 @@ Example:
 		interrupt-parent = <&gic>;
 		interrupts = <0 104 4>, <0 105 4>;
 		reg = <0x0 0x80010000 0x0 0x1000>;
+		xlnx,tx = <&i2s_transmitter>;
+		xlnx,rx = <&i2s_receiver>;
 		clock-names = "s_axi_lite_aclk", "aud_mclk";
 		clocks = <&clk 71>, <&clk_wiz_1 0>;
 	};
-- 
2.31.1

